backend/program/realdebrid.py

```diff
@@ -146,27 +138,39 @@ def _set_file_paths(self, item):
             self._handle_episode_paths(item)
 
     def _handle_movie_paths(self, item):
-        item.set("folder", item.active_stream.get("name"))
-        item.set("alternative_folder", item.active_stream.get("alternative_name"))
-        item.set("file", next(file for file in item.active_stream.get("files").values())["filename"])
+        wanted_files = self._filter_wanted_files(item.active_stream.get("files"))
+        if wanted_files:
+            item.set("folder", item.active_stream.get("name"))
+            item.set("alternative_folder", item.active_stream.get("alternative_name"))
+            item.set("file", next(file_info["filename"] for file_info in wanted_files.values()))
 
     def _handle_season_paths(self, season):
-        for file in season.active_stream["files"].values():
-            for episode in parser.episodes_in_season(season.number, file["filename"]):
+        for file_info in self._filter_wanted_files(season.active_stream["files"]).values():
+            for episode in parser.episodes_in_season(season.number, file_info["filename"]):
                 if episode - 1 in range(len(season.episodes)):
-                    season.episodes[episode - 1].set(
-                        "folder", season.active_stream.get("name")
-                    )
-                    season.episodes[episode - 1].set(
-                        "alternative_folder", season.active_stream.get("alternative_name")
-                    )
-                    season.episodes[episode - 1].set("file", file["filename"])
+                    season.episodes[episode - 1].set("folder", season.active_stream.get("name"))
+                    season.episodes[episode - 1].set("alternative_folder", season.active_stream.get("alternative_name"))
+                    season.episodes[episode - 1].set("file", file_info["filename"])
 
     def _handle_episode_paths(self, episode):
-        file = next(file for file in episode.active_stream.get("files").values() if episode.number in parser.episodes_in_season(episode.parent.number, file["filename"]))
-        episode.set("folder", episode.active_stream.get("name"))
-        episode.set("alternative_folder", episode.active_stream.get("alternative_name"))
-        episode.set("file", file["filename"])
+        wanted_files = self._filter_wanted_files(episode.active_stream.get("files"))
+        if wanted_files:
+            file_info = next(file_info for file_info in wanted_files.values() if episode.number in parser.episodes_in_season(episode.parent.number, file_info["filename"]))
+            episode.set("folder", episode.active_stream.get("name"))
+            episode.set("alternative_folder", episode.active_stream.get("alternative_name"))
+            episode.set("file", file_info["filename"])
+
+    def _filter_wanted_files(self, files):
+        """Filter unwanted files based on file extension and exclude 'sample' variations"""
+        unwanted_keywords = ["sample"]
```

Comment:
should do by size instead.. theres too many keywords we'd need to match.. better off cutting off anything under X filesize, which can be set by user. 100mb default.